packages/compiler/src/server/yaml-resolver.ts

```diff
@@ -251,32 +304,43 @@ function createYamlPathFromVisitScalarNode(
       // if the scalar node is marked as value but separated by newline from the key, it's more likely that the user is inputting the first property of an object
       // so build the target as an object key
       path.push(n.source ?? "");
+
       return {
         path,
         type: "key",
         source: n.source ?? "",
+        sourceType: n.type ?? "",
         siblings: [],
+        parameters: configParams,
+        envs: configEnvs,
       };
     } else {
-      const parent = nodePath[nodePath.length - 2];
+      const parent = nodePath.length >= 2 ? nodePath[nodePath.length - 2] : undefined;
       const targetSiblings = isMap(parent)
         ? parent.items.filter((item) => item !== last).map((item) => (item.key as any).source ?? "")
         : [];
+
       return {
         path: path,
         type: key === "key" ? "key" : "value",
         source: n.source ?? "",
         siblings: targetSiblings,
+        sourceType: n.type ?? "",
+        parameters: configParams,
+        envs: configEnvs,
       };
     }
   } else if (isSeq(last)) {
     return {
       path: path,
       type: "arr-item",
       source: n.source ?? "",
+      sourceType: n.type ?? "",
       siblings: last.items
         .filter((i) => i !== n)
         .map((item) => (isScalar(item) ? (item.source ?? "") : "")),
+      parameters: configParams,
+      envs: configEnvs,
     };
   } else {
     log({
```

Comment:
please also add test accordingly, you can check existing test for completion as example